src/api/API.js

```diff
@@ -0,0 +1,114 @@
+import EventEmitter from 'events';
+import axios from 'axios';
+
+import Annotation from '../Annotation';
+import { getHeaders } from '../util';
+
+class API extends EventEmitter {
+    /**
+     * Generates a rfc4122v4-compliant GUID, from
+     * http://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript.
+     *
+     * @return {string} UUID for annotation
+     */
+    static generateID() {
+        /* eslint-disable */
+        return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (c) => {
+            var r = (Math.random() * 16) | 0,
+                v = c == 'x' ? r : (r & 0x3) | 0x8;
+            return v.toString(16);
+        });
+        /* eslint-enable */
+    }
+
+    /**
+     * [constructor]
+     *
+     * @param {Object} data - Annotation Service data
+     * @return {API} API instance
+     */
+    constructor(data) {
+        super();
+        this.apiHost = data.apiHost;
+        this.fileId = data.fileId;
+        this.headers = getHeaders({}, data.token);
+        this.canAnnotate = data.canAnnotate;
+        this.user = {
+            id: '0',
+            name: this.anonymousUserName
+        };
+
+        this.axios = axios.create();
+        this.axiosSource = axios.CancelToken.source();
+    }
+
+    makeRequest(methodRequest, successCallback, errorCallback) {
+        return new Promise((resolve, reject) => {
+            methodRequest
+                .then((data) => {
+                    successCallback(data.data);
+                    resolve(data.data);
+                })
+                .catch((error) => {
+                    errorCallback(error);
+                    reject(errorCallback);
+                });
+        });
+    }
+
+    getParsedUrl(url) {
+        const a = document.createElement('a');
```

Comment:
Is this logic/method necessary for the API to work? React generally advises not interacting directly with browser variables outside of `componentDid...` methods.

Reply:
This isn't actually adding anything into the DOM. This is modeled after the same method in elements https://github.com/box/box-ui-elements/blob/master/src/util/Xhr.js#L106

Reply:
My comment wasn't really restricted to adding things to the DOM, but to interacting with browser-specific global variables outside of certain lifecycle methods. I realize the code also exists in UI Elements, but is it necessary for things to work? If so, can it be parsed in a different way?

Reply:
Removed it altogether. It's not needed for any annotations calls as far as I can tell